Error: spurious trailing ‘%’ in format

I encountered this problem when compiling with gcc 5.3.0 and cuda 8.0.

```
/home/kmcuda/src/python.cc: In function ‘bool validate_features_size(uint32_t)’:
/home/kmcuda/src/python.cc:108:45: error: expected ‘)’ before ‘PRIu32’:
     sprintf(msg, "\"samples\": more than %" PRIu32 " features is not supported",
                                             ^~~~~~
/home/kmcuda/src/python.cc:109:26: error: spurious trailing ‘%’ in format [-Werror=format=]
```

[This page](https://stackoverflow.com/questions/8132399/how-to-printf-uint64-t-fails-with-spurious-trailing-in-format) explains why and by adding `#define __STDC_FORMAT_MACROS` at the begining of python.cc file I can compile successfully.

Also I tried gcc 6.3.0 and cuda 9.0 this problem still exists.

--- a/.travis.osx
+++ b/.travis.osx
@@ -2,7 +2,8 @@
 
 brew update
 brew install llvm --with-clang
-brew install python3 r p7zip
+brew install r p7zip
+brew upgrade python
 pip3 install numpy
 curl -O https://storage.googleapis.com/srcd-packages/cuda_8.0.61_mac.dmg
 7z x cuda_8.0.61_mac.dmg

--- a/src/python.cc
+++ b/src/python.cc
@@ -1,3 +1,6 @@
+/// avoid spurious trailing ‘%’ in format error
+/// see https://stackoverflow.com/questions/8132399/how-to-printf-uint64-t-fails-with-spurious-trailing-in-format
+#define __STDC_FORMAT_MACROS
 #include <memory>
 #include <string>
 #include <unordered_map>

